src/os/bluestore/Allocator.h

```diff
@@ -16,11 +16,186 @@
 #include <ostream>
 #include "include/ceph_assert.h"
 #include "bluestore_types.h"
+#include "common/ceph_mutex.h"
 
 typedef interval_set<uint64_t> release_set_t;
 typedef release_set_t::value_type release_set_entry_t;
 
 class Allocator {
+protected:
+
+  struct ExtentCollectionTraits {
```

Comment:
Only Btree2Allocator uses ExtentCollectionTraits and LocklessWillfulStack and WillfulExtentCache.
Maybe it all should go to some "Allocator-cache.h"?

Reply:
I'm planning to bring it to every allocator in the future hence moved it here.  So IMO it makes sense to leave this here